### DISCUSSION TITLE
is f_jit(x) different compared to jax.jit(f)(x)

### DISCUSSION THREAD

**Discussion by rozeappletree** [jax-ml/jax#8675] [Q&A] [answered]
```
selu_jit = jax.jit(selu)
%timeit -n1000 -r5 selu_jit(x).block_until_ready()
```
```
1000 loops, best of 5: 122 µs per loop
```

and 

```
%timeit -n1000 -r5 jax.jit(selu)(x).block_until_ready()
```
```
1000 loops, best of 5: 270 µs per loop
```

Is the difference significant

**Comment by jakevdp** [comment, marked as answer]
The difference here is the time that it takes the `jax.jit` transform to do a cache lookup on `selu` and construct & return a wrapped function. You can measure this more directly if you wish:
```
%timeit jax.jit(selu)
```
As for whether it's significant – it's a one-time cost of several tens of microseconds. When running microbenchmarks, this can be significant. If you're JIT-compiling a larger/more complex piece of code, it may not be significant compared to the cost of the full computation.
